src/main/java/com/microsoft/jfr/dcmd/FlightRecorderDiagnosticCommandConnection.java

```diff
@@ -106,6 +106,10 @@ public static FlightRecorderConnection connect(MBeanServerConnection mBeanServer
     @Override
     public long startRecording(RecordingOptions recordingOptions, RecordingConfiguration recordingConfiguration) throws JfrStreamingException {
```

Comment:
Compiler should complain about raw-type here, no? If you make this `RecordingConfiguration<String>`, then the `recordingConfiguration instanceof RecordingConfiguration.MapConfiguration` is unnecessary.

Reply:
> Compiler should complain about raw-type here, no? 
 */

 JDK 11 and JDK 17 compilers do not seem to complain about this.

> If you make this RecordingConfiguration<String>, then the recordingConfiguration instanceof RecordingConfiguration.MapConfiguration is unnecessary.

This does not compile because `FlightRecorderDiagnosticCommandConnection` extends `FlightRecorderConnection`.

Reply:
I have applied the changes that have led to other changes to make it compile and work.